tests/Unit/NumericalAlgorithms/DiscontinuousGalerkin/Test_MetricIdentityJacobian.cpp

```diff
@@ -33,81 +33,75 @@ using Affine = domain::CoordinateMaps::Affine;
-      mesh.number_of_grid_points(), 0.0};
-  DataVector buffer(mesh.number_of_grid_points());
-  std::array<Mesh<1>, Dim> meshes_1d{};
-  for (size_t i = 0; i < Dim; ++i) {
-    gsl::at(meshes_1d, i) = mesh.slice_through(i);
-  }
-  const Matrix identity{};
-
-  for (size_t i_hat = 0; i_hat < Dim; ++i_hat) {  // logical index
-    auto diff_matrices = make_array<Dim>(std::cref(identity));
-    gsl::at(diff_matrices, i_hat) =
-        Spectral::differentiation_matrix(gsl::at(meshes_1d, i_hat));
-    for (size_t i = 0; i < Dim; ++i) {  // inertial index
-      apply_matrices(make_not_null(&buffer), diff_matrices,
-                     result.get(i_hat, i), mesh.extents());
-      divergence_terms.get(i) += buffer;
+  {
+    tnsr::I<DataVector, Dim, Frame::Inertial> inertial_coords{
+        mesh.number_of_grid_points()};
+    Jacobian<DataVector, Dim, Frame::Logical, Frame::Inertial> jacobian{
+        mesh.number_of_grid_points()};
+
+    fill_with_random_values(make_not_null(&inertial_coords),
+                            make_not_null(&gen), make_not_null(&dist));
+    fill_with_random_values(make_not_null(&jacobian), make_not_null(&gen),
+                            make_not_null(&dist));
+
+    InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial> result{};
+
+    dg::metric_identity_det_jac_times_inv_jac(make_not_null(&result), mesh,
+                                              inertial_coords, jacobian);
+
+    // Check metric identities are satisfied by taking the numerical divergence
+    tnsr::i<DataVector, Dim, Frame::Inertial> divergence_terms{
+        mesh.number_of_grid_points(), 0.0};
+    DataVector buffer(mesh.number_of_grid_points());
+    std::array<Mesh<1>, Dim> meshes_1d{};
+    for (size_t i = 0; i < Dim; ++i) {
+      gsl::at(meshes_1d, i) = mesh.slice_through(i);
+    }
+    const Matrix identity{};
+
+    for (size_t i_hat = 0; i_hat < Dim; ++i_hat) {  // logical index
+      auto diff_matrices = make_array<Dim>(std::cref(identity));
+      gsl::at(diff_matrices, i_hat) =
+          Spectral::differentiation_matrix(gsl::at(meshes_1d, i_hat));
+      for (size_t i = 0; i < Dim; ++i) {  // inertial index
+        apply_matrices(make_not_null(&buffer), diff_matrices,
+                       result.get(i_hat, i), mesh.extents());
+        divergence_terms.get(i) += buffer;
+      }
     }
-  }
 
-  Approx local_approx = Approx::custom().epsilon(1.0e-12).scale(1.);
-  const DataVector expected{mesh.number_of_grid_points(), 0.0};
-  for (size_t i = 0; i < Dim; ++i) {
-    CHECK_ITERABLE_CUSTOM_APPROX(divergence_terms.get(i), expected,
-                                 local_approx);
+    Approx local_approx = Approx::custom().epsilon(1.0e-12).scale(1.);
+    const DataVector expected{mesh.number_of_grid_points(), 0.0};
+    for (size_t i = 0; i < Dim; ++i) {
+      CHECK_ITERABLE_CUSTOM_APPROX(divergence_terms.get(i), expected,
+                                   local_approx);
+    }
   }
```

Comment:
For my understanding: why did this first set of checks get moved into a scope?

Reply:
If remember correctly it was name collisions, e.g. `jacobian` and `inertial_corods` being reused but in the first case with random values and the second with "motivated" values